playbooks/tests/tasks/integration.yml

```diff
@@ -6,24 +6,50 @@
     user: name=root generate_ssh_key=yes
 
   - name: generate nova key-pair
-    shell: . /root/stackrc; nova keypair-add turtle-key --pub-key
-           /root/.ssh/id_rsa.pub
+    nova_keypair: name=turtle-key
+                  auth_url={{ endpoints.auth_uri }}
+                  login_tenant_name=admin
+                  login_username=provider_admin
+                  login_password={{ secrets.provider_admin_password }}
+                  public_key=/root/.ssh/id_rsa.pub
+    run_once: true
 
   - name: generate test security group
-    shell: . /root/stackrc; neutron security-group-create turtle-sec && 
+    shell: . /root/stackrc; neutron security-group-create turtle-sec &&
            neutron security-group-rule-create turtle-sec --remote-ip-prefix
            0.0.0.0/0
 
+  - name: determine Neutron internal interface ID
+    neutron_network: name=external
+                     auth_url={{ endpoints.auth_uri }}
+                     login_tenant_name=admin
+                     login_username=provider_admin
+                     login_password={{ secrets.provider_admin_password }}
+    register: internal_net
```

Comment:
You're discovering external network here, but registering it as the internal_net, which means you're booting the instance on the external network, which will not work :)  name= should be internal.